native/src/main/java/io/ballerina/stdlib/mcp/SseEventStreamHelper.java

```diff
@@ -58,13 +62,14 @@ public static void attachSseStream(BObject object, BStream sseStream) {
      *
      * @param env       The Ballerina runtime environment.
      * @param object    The Ballerina object holding the SSE stream as native data.
-     * @return          The next SSE event record, or null if the stream is exhausted.
-     * @throws IllegalStateException if the SSE stream has not been attached to the object.
+     * @return          The next SSE event record, null if the stream is exhausted,
+     *                  or a Ballerina error object if unavailable.
      */
     public static Object getNextSseEvent(Environment env, BObject object) {
         BStream sseStream = (BStream) object.getNativeData(SSE_STREAM_NATIVE_KEY);
         if (sseStream == null) {
-            throw new IllegalStateException("SSE stream not attached to the provided Ballerina object.");
+            BString errorMessage = fromString("SSE stream not attached to the provided Ballerina object.");
```

Comment:
We can change the error message here `Unable to obtain elements from stream. SSE stream not found.`